patches/net/minecraft/client/resources/SplashManager.java.patch

```diff
@@ -0,0 +1,44 @@
+--- a/net/minecraft/client/resources/SplashManager.java
++++ b/net/minecraft/client/resources/SplashManager.java
+@@ -34,12 +_,12 @@
+ 
+     protected List<String> prepare(ResourceManager p_118869_, ProfilerFiller p_118870_) {
+         try {
+-            List list;
++            List<String> list;
+             try (BufferedReader bufferedreader = Minecraft.getInstance().getResourceManager().openAsReader(SPLASHES_LOCATION)) {
+                 list = bufferedreader.lines().map(String::trim).filter(p_118876_ -> p_118876_.hashCode() != 125780783).collect(Collectors.toList());
+             }
+ 
+-            return list;
++            return net.neoforged.neoforge.client.resources.NeoSplashHooks.prepare(list, p_118869_, p_118870_);
+         } catch (IOException ioexception) {
+             return Collections.emptyList();
+         }
+@@ -52,20 +_,12 @@
+ 
+     @Nullable
+     public SplashRenderer getSplash() {
+-        Calendar calendar = Calendar.getInstance();
+-        calendar.setTime(new Date());
+-        if (calendar.get(2) + 1 == 12 && calendar.get(5) == 24) {
+-            return SplashRenderer.CHRISTMAS;
+-        } else if (calendar.get(2) + 1 == 1 && calendar.get(5) == 1) {
+-            return SplashRenderer.NEW_YEAR;
+-        } else if (calendar.get(2) + 1 == 10 && calendar.get(5) == 31) {
+-            return SplashRenderer.HALLOWEEN;
+-        } else if (this.splashes.isEmpty()) {
++        List<String> splashes = net.neoforged.neoforge.client.resources.NeoSplashHooks.getSpecialSplashes();
++        if (splashes == null)
++            splashes = this.splashes;
++        if (splashes.isEmpty())
+             return null;
+-        } else {
+-            return this.user != null && RANDOM.nextInt(this.splashes.size()) == 42
+-                ? new SplashRenderer(this.user.getName().toUpperCase(Locale.ROOT) + " IS YOU")
+-                : new SplashRenderer(this.splashes.get(RANDOM.nextInt(this.splashes.size())));
+-        }
++        String name = user.getName();
```

Comment:
Given the presence of a null check in vanilla, I'd still keep it just in case.
We cannot really rely on nullable annotations for vanilla because sometimes stuff could be null even if not marked as such in my experience.